deploy/eck-stack/charts/eck-kibana/templates/kibana.yaml

```diff
@@ -12,4 +12,49 @@ metadata:
     {{- end }}
 spec:
   version: {{ required "A Kibana version is required" .Values.version }}
-  {{- toYaml .Values.spec | nindent 2 }}
+  {{- /* 
+    The following templates with 'or' are to allow both .spec.field and .field to be set for backwards 
+    compatibility purposes. See https://github.com/elastic/cloud-on-k8s/pull/8192 for details.
+  */ -}}
+  {{- with or ((.Values.spec).image) (.Values.image) }}
+  image: {{ . }}
+  {{- end }}
+  {{- with or ((.Values.spec).count) (.Values.count) }}
+  count: {{ . }}
+  {{- end }}
+  {{- $esRef := or ((.Values.spec).elasticsearchRef) (.Values.elasticsearchRef) }}
+  {{- if not ($esRef).name }}
+  {{ fail "An elasticsearchRef is required" }}
+  {{- end }}
+  elasticsearchRef:
+  {{- toYaml $esRef | nindent 4 }}
+  {{- $entsearchRef := or ((.Values.spec).enterpriseSearchRef) (.Values.enterpriseSearchRef) }}
+  {{- if $entsearchRef }}
+  enterpriseSearchRef:
+  {{- toYaml $entsearchRef | nindent 4 }}
+  {{- end }}
+  {{- with or ((.Values.spec).config) (.Values.config) }}
+  config:
+    {{ toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with or ((.Values.spec).http) (.Values.http) }}
+  http:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with or ((.Values.spec).podTemplate) (.Values.podTemplate) }}
+  podTemplate:
+    {{ toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with or ((.Values.spec).revisionHistoryLimit) (.Values.revisionHistoryLimit) }}
+  revisionHistoryLimit: {{ . }}
+  {{- end }}
+  {{- with or ((.Values.spec).secureSettings) (.Values.secureSettings) }}
+  secureSettings:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with or ((.Values.spec).serviceAccountName) (.Values.serviceAccountName) }}
+  serviceAccountName: {{ . }}
+  {{- end }}
+  {{- with or ((.Values.spec).monitoring) (.Values.monitoring) }}
+  monitoring: {{ . }}
```

Comment:
```suggestion
  monitoring:
    {{- toYaml . | nindent 4 }}
```

Reply:
Let me actually get a test in place, as this should have been caught...

Reply:
Ok, new test showed the issue.
```
        - values.spec.monitoring should render properly

                - asserts[1] `equal` fail
                        Template:       eck-kibana/templates/kibana.yaml
                        DocumentIndex:  0
                        Path:   spec.monitoring
                        Expected to equal:
                                logs:
                                  elasticsearchRefs:
                                  - name: monitoring
                                    namespace: observability
                                metrics:
                                  elasticsearchRefs:
                                  - name: monitoring
                                    namespace: observability
                        Actual:
                                map[logs:map[elasticsearchRefs:[map[name:monitoring namespace:observability]]] metrics:map[elasticsearchRefs:[map[name:monitoring
                                  namespace:observability]]]]
```

And then fixed it

```
❯ helm unittest -3 -f 'templates/tests/*.yaml' 

### Chart [ eck-kibana ] .

 PASS  test kibana ingress      templates/tests/ingress_test.yaml
 PASS  test kibana      templates/tests/kibana_test.yaml

Charts:      1 passed, 1 total
Test Suites: 2 passed, 2 total
Tests:       21 passed, 21 total
Snapshot:    0 passed, 0 total
Time:        48.881008ms
```